Marlin/src/lcd/menu/menu_tmc.cpp

```diff
@@ -78,50 +78,50 @@ void menu_tmc_current() {
   START_MENU();
   MENU_BACK(MSG_TMC_DRIVERS);
   #if AXIS_IS_TMC(X)
-    TMC_EDIT_STORED_I_RMS(X);
+    TMC_EDIT_STORED_I_RMS(X, MSG_X);
   #endif
   #if AXIS_IS_TMC(Y)
-    TMC_EDIT_STORED_I_RMS(Y);
+    TMC_EDIT_STORED_I_RMS(Y, MSG_Y);
   #endif
   #if AXIS_IS_TMC(Z)
-    TMC_EDIT_STORED_I_RMS(Z);
+    TMC_EDIT_STORED_I_RMS(Z, MSG_Z);
   #endif
   #if AXIS_IS_TMC(X2)
-    TMC_EDIT_STORED_I_RMS(X2);
+    TMC_EDIT_STORED_I_RMS(X2, MSG_X2);
   #endif
   #if AXIS_IS_TMC(Y2)
-    TMC_EDIT_STORED_I_RMS(Y2);
+    TMC_EDIT_STORED_I_RMS(Y2, MSG_Y2);
   #endif
   #if AXIS_IS_TMC(Z2)
-    TMC_EDIT_STORED_I_RMS(Z2);
+    TMC_EDIT_STORED_I_RMS(Z2, MSG_Z2);
   #endif
   #if AXIS_IS_TMC(Z3)
-    TMC_EDIT_STORED_I_RMS(Z3);
+    TMC_EDIT_STORED_I_RMS(Z3, MSG_Z3);
   #endif
   #if AXIS_IS_TMC(E0)
-    TMC_EDIT_STORED_I_RMS(E0);
+    TMC_EDIT_STORED_I_RMS(E0, MSG_E1);
```

Comment:
The output for this low-level stepper designation should remain zero-based, so changes are not needed here.

Reply:
@thinkyhead: I made this change for consistency with what you are doing [here](https://github.com/MarlinFirmware/Marlin/blob/04ae3ba1fdf813fee2431b4626af6aa5cf1b1b87/Marlin/src/module/temperature.cpp#L98):

```
#define _E_PSTR(M,E,N) ((HOTENDS) >= (N) && (E) == (N)-1) ? PSTR(MSG_E##N " " M) :
#define TEMP_ERR_PSTR(M,E) _BED_PSTR(M##_BED,E) _CHAMBER_PSTR(M##_CHAMBER,E) _E_PSTR(M,E,2) _E_PSTR(M,E,3) _E_PSTR(M,E,4) _E_PSTR(M,E,5) _E_PSTR(M,E,6) PSTR(MSG_E1 " " M)
```

There, if e == 0, the MSG_E1 is printed; if e == 1, then MSG_E2 is printed... etc.

So, I modified the TCM macro to do the same thing:

```
TMC_EDIT_STORED_I_RMS(E0, MSG_E1);
````

When e == 0 (i.e. E0), MSG_E1 is printed.

> The output for this low-level stepper designation should remain zero-based, so changes are not needed here.

I disagree. You can't have it one way in part of the code and another way in another part of the code.

Reply:
BTW, those macros are for LCD menus. On re-reading your comment, it seems like you may have missed that point.

Reply:
Why "low level"? ... this is shown in the LCD.

Reply:
To be honest, I don't care as much about the TMC menu, as it's hidden fairly deep, but in our FW I've been asked to replace all user facing strings of "E0/E1" with "Hot End 1" and "Hot End 2" for dual extruders and with "Hot End" for single extruder machines. I generally accomplish this by doing the following in "core/language.h":

```
#if EXTRUDERS == 1
  #define MSG_E1 "Hot End"
  #define MSG_E2 "Hot End"
#else if EXTRUDERS == 2
  #define MSG_E1 "Hot End 1"
  #define MSG_E2 "Hot End 2"
#endif
```

But according to what you just said, certain places you want MSG_E1 to be a user facing string, and certain other places you want it to be a "low-level stepper designation" ... but if you are using the exact same constant, then there is no way for me to actually modify these strings in a consistent way. The best I can do is the following:

```
#if EXTRUDERS == 1
  #define MSG_E0 "Hot End"
  #define MSG_E1 "Hot End"
  #define MSG_E2 "Hot End"
#else if EXTRUDERS == 2
  #define MSG_E0 "Hot End 0"
  #define MSG_E1 "Hot End 1"
  #define MSG_E2 "Hot End 2"
#endif
```

But that means that the "TMC menu" will now show "X, Y, Z, Hot End 0, Hot End 1" while the other menus, such as Move, will show "X, Y, Z, Hot End 1, Hot End 2". This is quite sloppy...

Reply:
I was about to remove `MSG_E0` completely the other day.

Reply:
Sorry, I conflated the TMC menu code with TMC debug / util code. Right, so E1 should be the user-facing designation on the LCD. I would discourage using "Hot End " in place of "E" because various menu strings will likely become too long.